pkg/utils/sid.go

```diff
@@ -1,18 +1,28 @@
 package utils
 
+// Operations to marshall to/from SIDs.
+// It is recommended not to use these directly - instead use a wrapper type that
+// performs the relevant error handling.
+
 const (
-	nodeIDMask  uint64 = 0xFFFF << 48
+	nodeIDBits  uint64 = 16
```

Comment:
It would be good to have some really clear comments around how these IDs are supposed to work, since bitmasks are not the most readable.